core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -129,13 +131,33 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
+        ctrl.saveBeforeUnload();
+      };
+
+      $scope.canPublish = false;
 
+      $scope.showPublishButton = function() {
```

Comment:
As per offline discussion:

- Attach `explorationCanBePublished` to ng-if and update `explorationCanBePublished` in the promise callback.

Reply:
Will have to work on this later. Maybe a separate PR.

Reply:
Sorry, I'm finding it hard to understand the last comment, can you please rephrase it?

Reply:
I had implemented an event emitter to broadcast a message when the exploration is Unpublished. Then I had attached the `explorationCanBePublished` to the ng-if but this broke something and the `publish` button stopped appearing in the navbar.
I'm planning to take a look at it in a separate PR

Reply:
> but this broke something

Can you ad detail on this?

> I'm planning to take a look at it in a separate PR

Please don't move this to a separate PR, this is related to the changes for this PR and I would suggest doing it properly in the first place.

Reply:
Fixed